Cargo.lock

```diff
@@ -2489,12 +2487,50 @@ version = "0.3.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "c08d65885ee38876c4f86fa503fb49d7b507c2b62552df7c70b2fce627e06381"
 
+[[package]]
+name = "openssl"
+version = "0.10.64"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "95a0481286a310808298130d22dd1fef0fa571e05a8f44ec801801e84b216b1f"
+dependencies = [
+ "bitflags 2.4.2",
+ "cfg-if",
+ "foreign-types",
+ "libc",
+ "once_cell",
+ "openssl-macros",
+ "openssl-sys",
+]
+
+[[package]]
+name = "openssl-macros"
+version = "0.1.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "a948666b637a0f465e8564c73e89d4dde00d72d4d473cc972f390fc3dcee7d9c"
+dependencies = [
+ "proc-macro2",
+ "quote",
+ "syn 2.0.52",
+]
+
 [[package]]
 name = "openssl-probe"
 version = "0.1.5"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "ff011a302c396a5197692431fc1948019154afc178baf7d8e37367442a4601cf"
 
+[[package]]
+name = "openssl-sys"
+version = "0.9.102"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c597637d56fbc83893a35eb0dd04b2b8e7a50c91e64e9493e398b5df4fb45fa2"
+dependencies = [
+ "cc",
+ "libc",
+ "pkg-config",
+ "vcpkg",
+]
+
```

Comment:
I'd really like to avoid introducing a dependency like this

Reply:
Hmm, ok. But what can we do? @LecrisUT only bumped the metrics version.

I am just wondering, because I am not sure I understand. At one point the metrics crate in the current version might not be supported anymore. Or introduces a security issue in the future. 

I am not saying it has to be updated, but I am at a loss when it comes to updating dependencies.

Reply:
We can

1. Help upstream metrics introduce a feature so it does not require openssl
2. Fork upstream
3. Hold off on updating metrics because the current version is currently supported, and does not have security issues, perf issues, or any other reason to put the effort into (1) or (2). 

This exact issue illustrates my concern with the dependencies discussion in #2061. It's a really nice ideal, but does not seem practical given limited resources.

Reply:
Is it `metrics` or `metrics-exporter-prometheus`, and what's the chain of dependencies (and features?) that affects this? I think for the cross-compilation, it's just an issue on the environment side to use appropriate packages for cross-compilation.

Reply:
Thanks for the explanation. I think option 2 is not sustainable, since that might come up for other crates as well. So I think option 3 is the most feasible right now. Maybe I can find the time to look into that upstream issue.

Reply:
Also, I am ok with option 3 as long as we keep this PR open to track any subsequent necessary changes so that I can port them over.

Reply:
> I'm not ok with having a dependency on both rustls and openssl, nor do I want to switch to openssl just to update this crate.

Sure, option 3 is ok on my side. Just trying to get a grip of the issue, to know what to track.

> Do you mean that in Fedora Atuin will require openssl?

It's rather `hyper-tls` is pre-built with all necessary dependencies. I can ask the main rust packagers if you think there are issues that you want to avoid. Looking at the implementation of [Fedora's `hyper-tls`](https://src.fedoraproject.org/rpms/rust-hyper-tls/blob/rawhide/f/rust-hyper-tls.spec#_16-18) it seems it is dropping `rustls`?
```spec
# Manually created patch for downstream crate metadata changes
# * drop feature for using a vendored TLS backend
Patch:          hyper-tls-fix-metadata.diff
```

Reply:
> Do I then need to consider that some Atuin users will be affected in the case of a new CVE or other such issue?

This we are always tracking, and when a dependency has any changes, the whole dependency tree upwards is rebuilt.